main/modes/mode_slide_whistle.c

```diff
@@ -1031,14 +1141,31 @@ void  slideWhistleExitMode(void)
     freeFont(&slideWhistle->mm);
 
     esp_timer_stop(slideWhistle->beatTimer);
-    esp_timer_stop(slideWhistle->paramSwitchTimer);
 
     esp_timer_delete(slideWhistle->beatTimer);
-    esp_timer_delete(slideWhistle->paramSwitchTimer);
 
     free(slideWhistle);
 }
 
+/**
+ * This function is called when a button press is pressed. Buttons are
+ * handled by interrupts and queued up for this callback, so there are no
+ * strict timing restrictions for this function.
+ *
+ * @param evt The button event that occurred
+ */
+void  slideWhistleTouchCallback(touch_event_t* evt)
+{
+    slideWhistle->touchHeld = evt->state != 0;
+    slideWhistle->shouldPlay = evt->state != 0 || slideWhistle->aHeld;
+
+    //slideWhistle->touchPosition = roundf((evt->position * BAR_X_WIDTH) / 255);
+    getTouchCentroid(&slideWhistle->touchPosition, &slideWhistle->touchIntensity);
```

Comment:
I thought that calling this from the call gave choppy results?

Reply:
Might be better if you call this either in the main loop or the accel callback

Reply:
Fixed this myself

Reply:
Thanks! Meant to get around to it today, but you beat me to it :)